src/lpython/python_evaluator.cpp

```diff
@@ -28,14 +33,144 @@ PythonCompiler::PythonCompiler(CompilerOptions compiler_options)
     e{std::make_unique<LLVMEvaluator>()},
     eval_count{0},
 #endif
-    compiler_options{compiler_options}
-//    symbol_table{nullptr}
+    compiler_options{compiler_options},
+    symbol_table{nullptr}
 {
 }
 
 PythonCompiler::~PythonCompiler() = default;
 
 
+Result<PythonCompiler::EvalResult> PythonCompiler::evaluate(
+#ifdef HAVE_LFORTRAN_LLVM
+            const std::string &code_orig, bool verbose, LocationManager &lm,
+            LCompilers::PassManager& pass_manager, diag::Diagnostics &diagnostics
+#else
+            const std::string &/*code_orig*/, bool /*verbose*/,
+                LocationManager &/*lm*/, LCompilers::PassManager& /*pass_manager*/,
+                diag::Diagnostics &/*diagnostics*/
+#endif
+            )
+{
+#ifdef HAVE_LFORTRAN_LLVM
+    EvalResult result;
+    result.type = EvalResult::none;
+
+    // Src -> AST
+    Result<LCompilers::LPython::AST::ast_t*> res = get_ast2(code_orig, diagnostics);
+    LCompilers::LPython::AST::ast_t* ast;
+    if (res.ok) {
+        ast = res.result;
+    } else {
+        return res.error;
+    }
+
+    if (verbose) {
+        result.ast = LCompilers::LPython::pickle_python(*ast, true, true);
+    }
+
+    // AST -> ASR
+    Result<ASR::TranslationUnit_t*> res2 = get_asr3(*ast, diagnostics, lm);
+    ASR::TranslationUnit_t* asr;
+    if (res2.ok) {
+        asr = res2.result;
+    } else {
+        LCOMPILERS_ASSERT(diagnostics.has_error())
+        return res2.error;
+    }
+
+    if (verbose) {
+        result.asr = pickle(*asr, true, true, true);
+    }
+
+    // ASR -> LLVM
+    Result<std::unique_ptr<LLVMModule>> res3 = get_llvm3(*asr,
+        pass_manager, diagnostics, lm.files.back().in_filename);
+    std::unique_ptr<LCompilers::LLVMModule> m;
+    if (res3.ok) {
+        m = std::move(res3.result);
+    } else {
+        LCOMPILERS_ASSERT(diagnostics.has_error())
+        return res3.error;
+    }
+
+    if (verbose) {
+        result.llvm_ir = m->str();
+    }
+
+    bool call_init = false;
+    bool call_stmts = false;
+    std::string init_fn = "__module___main_____main____lpython_interactive_init_" + std::to_string(eval_count) + "__";
+    std::string stmts_fn = "__module___main_____main____lpython_interactive_stmts_" + std::to_string(eval_count) + "__";
```

Comment:
Construct this like

```
std::string base_fn_name = "_main_____main____lpython_interactive";
std::string init_fn = base_fn_name + "_init_" + std::to_string(eval_count);
std::string module_init_fn = "__module__" + init_fn;
```

Similarly for `stmts`. Now this can be reused when erasing symbol below.